cmd/minishift/cmd/start.go

```diff
@@ -127,6 +128,12 @@ func runStart(cmd *cobra.Command, args []string) {
 	setOcProxy()
 	setShellProxy()
 
+	if viper.IsSet("http-proxy") {
```

Comment:
I think this should be encapsulated into a function. Also what makes you think that you can use https proxy url if the http proxy is not set. What type of proxy does the subscription-manager expect? I would think it does https, right? So if you are in a system where two proxies are setup, one for HTTP and one for HTTPS, I would expect that you want to give the HTTPS proxy preference. Than again, I am not sure. It comes down to subscription-manager.